src/APIM_ARMTemplate/apimtemplate/Common/Templates/Logger/LoggerTemplateUtils.cs

```diff
@@ -0,0 +1,45 @@
+﻿using apimtemplate.Common.Constants;
+using apimtemplate.Common.Helpers;
+using apimtemplate.Common.Templates.Abstractions;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace apimtemplate.Common.Templates.Logger
+{
+
+    internal static class LoggerTemplateUtils
+    {
+        public static Dictionary<string, string> GetAllLoggerResourceIds(List<TemplateResource> resources)
+        {
+            Dictionary<string, string> logResIds = new Dictionary<string, string>();
+            foreach (LoggerTemplateResource resource in resources)
+            {
+                string validLoggerName = GetValidLoggerParamName(resource.name);
+                string resourceId = resource.properties.resourceId;
+                logResIds.Add(validLoggerName, resourceId);
+            }
+            return logResIds;
+        }
+
+        public static Template SetLoggerResourceId(Template loggerTemplate)
+        {
+            TemplateResource[] loggerResources = loggerTemplate.resources.ToArray();
```

Comment:
What if `loggerTemplate` or `resources` is null?

Reply:
only moved the code, didn't refactor it by now